solutions/usaco-1037.mdx

```diff
@@ -14,6 +14,95 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+void setIO(string name = "") {
+	ios_base::sync_with_stdio(0);
+	cin.tie(0);
+	if (name.size()) {
+		freopen((name + ".in").c_str(), "r", stdin);
+		freopen((name + ".out").c_str(), "w", stdout);
+	}
+}
+
+struct Shake {
+	int t, x, y;
+	bool operator<(const Shake& tmp) { return t < tmp.t; }
+};
+
+int main() {
+	setIO("tracing");
+
+	int n, t;
+	string s;
+	cin >> n >> t >> s;
+
+	vector<Shake> shakes(t);
+	vector<bool> cows_infected(n);
+	unordered_set<int> possible_patient;
+	int min_k = t, max_k = -1;
+
+	// checks if pair {patient_zero, k} is consistent with handshakes data over time
+	auto is_consistent = [&](int patient_zero, int k) {
+		vector<bool> tmp_infected(n);
+		vector<int> num_shakes(n);
+		tmp_infected[patient_zero] = true;
+		for (Shake sh : shakes) {
+			if (tmp_infected[sh.x]) {
+				num_shakes[sh.x]++;
+			}
+			if (tmp_infected[sh.y]) {
+				num_shakes[sh.y]++;
+			}
+			if (num_shakes[sh.x] <= k && tmp_infected[sh.x]) {
+				tmp_infected[sh.y] = true;
+			}
+			if (num_shakes[sh.y] <= k && tmp_infected[sh.y]) {
+				tmp_infected[sh.x] = true;
+			}
+		}
+		for (int i = 0; i < n; i++) {
+			if (tmp_infected[i] != cows_infected[i]) {
+				return false;
+			}
+		}
+		return true;
+	};
+
+	// marks infected cows
+	for (int x = 0; x < n; x++) {
```

Comment:
why `x`?

Reply:
Just a variable to loop through all chars of s;

Reply:
Any thoughts about it?

Reply:
why not just `i` lol

Reply:
I think naming it `i` or `x` wouldn't make a difference. Just an iterator.

Reply:
but `i` is more standard 🤔 
if theres no real reason id recommend just changing it back

Reply:
You are right. I updated it.